File: javaconstructs/Truck.java
interface Machine
{
    
    int distanceTravelled=100;
    public int getDistance();
}
interface Vehicle
{
    int  velocity=50;
    public int getVelocity();
}
class Plane implements Machine, Vehicle
{
    int time;
    int velocity;
    int distanceTravelled;
    
    
    public Plane(int velocity, int time)
    {
        this.velocity = velocity;
        this.time = time;
    }
    
    public int getDistance()
    {
        distanceTravelled= velocity*time; 
        System.out.println("Total Distance  is : "+distanceTravelled);
        return distanceTravelled;
    }
    public int getVelocity()
    {
        int velocity=distanceTravelled/time;
        System.out.println("Velocity  is : "+ velocity);
        return velocity;
    }
}
public class Truck implements Machine, Vehicle
{
    int time;
    int velocity;
    int distanceTravelled;
    
    
    public Truck(int velocity, int time)
    {
        this.velocity = velocity;
        this.time = time;
    }
    
    public int getDistance()
    {
        distanceTravelled= velocity*time; 
        System.out.println("Total Distance  is : "+distanceTravelled);
        return distanceTravelled;
    }
    public int getVelocity()
    {
        int velocity=distanceTravelled/time;
        System.out.println("Velocity  is : "+ velocity);
        return velocity;
    }
    public static void main(String args[])
    {
        Truck truck = new Truck(50,2);
        truck.getDistance();
        truck.getVelocity();
		
		Plane plane = new Plane(1000,3);
        plane.getDistance();
        plane.getVelocity();
    }
}

File: javavsjavascript/JavaScript/truck.js
function inheritsMultipleObjects(baseObject, superObjects) {
  return new Proxy(
    baseObject,
    {
      get(target, key, rec) {
        if (Reflect.ownKeys(target).includes(key)) {
          return Reflect.get(target, key);
        }
        const parent = superObjects.find(
          _parent => Reflect.has(_parent, key)
        );
        if (parent !== undefined) {
          return Reflect.get(parent, key);
        }
        return undefined;
      },
      has(target, key) {
        if (Reflect.ownKeys(target).includes(key)) {
          return true;
        }
        const parentHasKey = superObjects.some(
          _parent => Reflect.has(_parent, key)
        );
        if (parentHasKey) {
          return true;
        }
        return false;
      }
    }
  );
}

class Vehicle {
  getDistance() {
    return 100;
  }
}
class Machine {
  getVelocity() {
    return 50;
  }
}
class Truck {}
const _truck = new Truck();
const truck = inheritsMultipleObjects(
  _truck,
  [Truck.prototype,Vehicle.prototype, Machine.prototype]
);
console.log(truck.getDistance()); 
console.log(truck.getVelocity()); 
console.log(truck.constructor); 